components/chainhook-sdk/src/indexer/stacks/mod.rs

```diff
@@ -562,6 +591,198 @@ pub fn standardize_stacks_microblock_trail(
+                        chain_id: block_rejection.chain_id,
+                        signature: block_rejection.signature.to_hex(),
+                    }),
+                ),
+            },
+            SignerMessage::BlockPushed(nakamoto_block) => {
+                StacksSignerMessage::BlockPushed(BlockPushedData {
+                    block: standardize_stacks_nakamoto_block(&nakamoto_block),
+                })
+            }
+            SignerMessage::MockSignature(_) => StacksSignerMessage::MockSignature,
+            SignerMessage::MockProposal(_) => StacksSignerMessage::MockProposal,
+            SignerMessage::MockBlock(_) => StacksSignerMessage::MockBlock,
+        };
+        parsed_chunks.push(StacksStackerDbChunk {
+            contract: contract_id.clone(),
+            sig: slot.sig.clone(),
+            pubkey: get_signer_pubkey_from_stackerdb_chunk_slot(slot, &data_bytes)?,
+            message,
+        });
+    }
+
+    Ok(parsed_chunks)
+}
+
+#[cfg(feature = "stacks-signers")]
+pub fn standardize_stacks_nakamoto_block(block: &NakamotoBlock) -> NakamotoBlockData {
+    use miniscript::bitcoin::hex::Case;
+    use miniscript::bitcoin::hex::DisplayHex;
+
+    NakamotoBlockData {
+        header: NakamotoBlockHeaderData {
+            version: block.header.version,
+            chain_length: block.header.chain_length,
+            burn_spent: block.header.burn_spent,
+            consensus_hash: block.header.consensus_hash.to_hex(),
+            parent_block_id: block.header.parent_block_id.to_hex(),
+            tx_merkle_root: block.header.tx_merkle_root.to_hex(),
+            state_index_root: block.header.state_index_root.to_hex(),
+            timestamp: block.header.timestamp,
+            miner_signature: block.header.miner_signature.to_hex(),
+            signer_signature: block
+                .header
+                .signer_signature
+                .iter()
+                .map(|s| s.to_hex())
+                .collect(),
+            pox_treatment: block
+                .header
+                .pox_treatment
+                .serialize_to_vec()
+                .to_hex_string(Case::Lower),
+        },
+        // FIXME: Should we parse these?
+        transactions: vec![],
+    }
+}
+
+#[cfg(feature = "stacks-signers")]
+pub fn get_signer_pubkey_from_stackerdb_chunk_slot(
+    slot: &NewSignerModifiedSlot,
+    data_bytes: &Vec<u8>,
+) -> Result<String, String> {
+    use clarity::util::hash::Sha512Trunc256Sum;
+    use miniscript::bitcoin::{
+        key::Secp256k1, secp256k1::{
+            ecdsa::{RecoverableSignature, RecoveryId},
+            Message,
+        }
+    };
+
+    let mut digest_bytes = slot.slot_id.to_be_bytes().to_vec();
+    digest_bytes.extend(slot.version.to_be_bytes().to_vec());
+    digest_bytes.extend(data_bytes.clone());
+    let digest = Sha512Trunc256Sum::from_data(&digest_bytes).to_bytes();
```

Comment:
`data_bytes` must be hashed first by itself, e.g. 
```ts
/** Get the digest to sign that authenticates this chunk data and metadata */
function authDigest(slot: ModifiedSlot): Buffer {
  const hasher = crypto.createHash('sha512-256');
  hasher.update(toU32BeBytes(slot.slot_id));
  hasher.update(toU32BeBytes(slot.slot_version));

  // Calculate the hash of the chunk bytes. This is the SHA512/256 hash of the data
  const dataHash = crypto.hash('sha512-256', Buffer.from(slot.data, 'hex'), 'buffer');
  hasher.update(dataHash);

  return hasher.digest();
}

```

Reply:
Thanks, done